snappi/snappicommon.py

```diff
@@ -120,7 +122,73 @@ def _decode(self, dict_object):
         raise NotImplementedError()
 
 
-class SnappiObject(SnappiBase):
+class SnappiValidator(object):
+
+    __slots__ = ()
+
+    def __init__(self) -> None:
+        pass
+
+    def validate_mac(self, mac):
+        if mac == None:
+            return False
+        for sep in [':', '-', '.', ' ']:
+            if sep in mac:
+                mac = mac.replace(sep, '')
+                break
+        pattern = re.compile(r'^[\d+a-f]{12}$', re.I)
```

Comment:
compile all regex as part of `__init__` and for each validation, please see if we can have an alternative one liner.

if there are multiple possible formats, let's stick to only one (as specified by API spec). we can always extend it to other formats if needed.